src/main/scala/com/fulcrumgenomics/bam/CollectErccMetrics.scala

```diff
@@ -0,0 +1,253 @@
+    |                                response metrics.
+    |  2. ercc_detailed_metrics.txt - gives a per-ERCC-transcript expected concentration and observed fragment count.
+    |  3. ercc_metrics.pdf - plots the expected concentration versus teh observed fragment count.
+    |
+    |Secondary and supplementary reads will be ignored.  A read pair mapping to an ERCC transcript is counted if both
+    |ends of the pair map to the same ERCC transcript.  A minimum mapping quality can be required for reads to be
+    |counted as mapped to an ERCC transcript.
+  """)
+class CollectErccMetrics
+( @arg(flag='i', doc="Input SAM or BAM file of aligned reads in coordinate order.") val input: PathToBam,
+  @arg(flag='m', doc="Tab-delimited file containing ERCC IDs and expected concentrations.") val metadata: FilePath,
+  @arg(flag='o', doc="Output prefix.") val output: PathPrefix,
+  @arg(flag='c', doc="Minimum # of counts required to include an ERCC transcript.") val minTranscriptCount: Int = 3,
+  @arg(flag='M', doc="The minimum mapping quality") val minimumMappingQuality: Int = 10
+) extends FgBioTool with LazyLogging {
+
+  Io.assertReadable(input)
+  Io.assertReadable(metadata)
+  Io.assertCanWriteFile(output)
+
+  private val ScriptPath = "com/fulcrumgenomics/bam/CollectErccMetrics.R"
+
+  override def execute(): Unit = {
+    val in       = SamSource(input)
+    val progress = ProgressLogger(logger)
+    val erccData = DelimitedDataParser(metadata, '\t').map { row => row[String](0) -> row[Double](1) }.toMap
+
+    // check for no ERCC data
+    if (erccData.isEmpty) {
+      throw new IllegalArgumentException(s"No ERCC data given in '$metadata'")
+    }
+
+    // require that the SAM/BAM header contains all the ERCC transcripts
+    erccData.keys.foreach { erccId =>
+      if (in.header.getSequence(erccId) == null) {
+        throw new IllegalArgumentException(s"ERCC with id '$erccId' not found in the SAM/BAM header")
+      }
+    }
+
+    var totalReads: Long    = 0 // the total # of reads in the BAM file
+    var erccReads: Long     = 0 // the total # of reads that are counted as mapped to an ERCC transcript
+    var erccFragments: Long = 0 // the total # of fragments or read pairs that are counted as mapped to an ERCC transcript
+    val erccFragmentCounter = new SimpleCounter[String]() // the count of fragments mapped to a given ERCC transcript
+    in.iterator
+      .filterNot { r => progress.record(r); r.secondary || r.supplementary } // no secondary or supplementary
+      .filter { r =>
+        totalReads += 1
+        // consider only records mapped to an ERCC transcript
+        r.mapped && r.mapq >= minimumMappingQuality && erccData.contains(r.refName)
+      }
+      .filter { r =>
+        erccReads += 1
+        // consider only paired records where both ends map to the sam ERCC transcript
+        r.paired && r.mapped && r.mateMapped && r.refIndex == r.mateRefIndex && r.firstOfPair
```

Comment:
Is there a good reason to only allow paired data?  I don't know enough to know whether there are folks still doing single end RNASeq.